theano/gof/tests/test_vm.py

```diff
@@ -212,14 +212,18 @@ def check_partial_function(linker_name):
     check_partial_function('cvm')
 
 
-# TODO: implement output_keys with CVM
-def test_partial_function_output_keys():
-    x = tensor.scalar('input')
-    y = 3 * x
-    f = theano.function([x], {'a': y * 5, 'b': y - 7}, mode=Mode(
-        optimizer=None, linker=vm.VM_Linker(allow_partial_eval=True)))
-
-    assert f(5, output_subset=['a'])['a'] == f(5)['a']
+def test_partial_function_with_output_keys():
+
+    def check_partial_function_output_keys(linker_name):
+        x = tensor.scalar('input')
+        y = 3 * x
+        f = theano.function([x], {'a': y * 5, 'b': y - 7}, mode=Mode(
+            optimizer=None, linker=linker_name))
+
+        assert f(5, output_subset=['a'])['a'] == f(5)['a']
+
+    check_partial_function_output_keys(vm.VM_Linker(allow_partial_eval=True))
```

Comment:
You should also add use_cloop=False to force the python version. Otherwise, with this PR, it will trigger the C code in both case.